tests/unit/src/test/scala/tests/codeactions/ExtractRenameMemberLspSuite.scala

```diff
@@ -0,0 +1,568 @@
+          |}
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedMember(
+    "extract-class-without-non-in-scope-imports",
+    """|package a
+       |import scala.io.Source
+       |
+       |case class A() {
+       |  import scala.concurrent.Future
+       |}
+       |case class <<B>>() {
+       |  val s = Source.fromFile("A.scala")
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("class", "B")}""".stripMargin,
+    """|package a
+       |import scala.io.Source
+       |
+       |case class A() {
+       |  import scala.concurrent.Future
+       |}
+       |""".stripMargin,
+    newFile = (
+      "B.scala",
+      s"""|package a
+          |
+          |import scala.io.Source
+          |
+          |case class B() {
+          |  val s = Source.fromFile("A.scala")
+          |}
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedMember(
+    "extract-class-inner-package",
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |case class <<B>>()
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("class", "B")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |
+       |}
+       |""".stripMargin,
+    newFile = (
+      "B.scala",
+      s"""|package a
+          |package b
+          |
+          |case class B()
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedMember(
+    "extract-class-inner-package-no-subpackages",
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |case class <<B>>()
+       |  package c {
+       |    case class C()
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("class", "B")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |
+       |  package c {
+       |    case class C()
+       |  }
+       |}
+       |""".stripMargin,
+    newFile = (
+      "B.scala",
+      s"""|package a
+          |package b
+          |
+          |case class B()
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedMember(
+    "extract-class-inner-subpackage",
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |  case class B()
+       |  package c {
+       |case class <<C>>()
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("class", "C")}""".stripMargin,
+    """|package a
+       |
+       |case class A()
+       |
+       |package b {
+       |  case class B()
+       |  package c {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    newFile = (
+      "C.scala",
+      s"""|package a
+          |package b
+          |package c
+          |
+          |case class C()
+          |""".stripMargin
+    )
+  )
+
+  checkExtractedMember(
+    "extract-class-inner-package-no-other-packages",
```

Comment:
Also, do we need this test? It seems to duplicate much of the other tests. I would honestly minimize the number of LSP tests as previously mentioned and just test the CodeAction itself for unit tests/